docs/en/engines/table-engines/mergetree-family/invertedindexes.md

```diff
@@ -60,70 +59,130 @@ ENGINE = MergeTree
-examples, consider:
+## Using a Text Index {#using-a-text-index}
+
+To use the index, no special functions or syntax are required.
+Typical string search predicates automatically leverage the index. As examples, consider:
 
 ```sql
-INSERT INTO tab(key, str) values (1, 'Hello World');
+INSERT INTO tab(key, str) VALUES (1, 'Hello World');
 SELECT * from tab WHERE str == 'Hello World';
 SELECT * from tab WHERE str IN ('Hello', 'World');
 SELECT * from tab WHERE str LIKE '%Hello%';
 SELECT * from tab WHERE multiSearchAny(str, ['Hello', 'World']);
 SELECT * from tab WHERE hasToken(str, 'Hello');
 ```
 
-The full-text index also works on columns of type `Array(String)`, `Array(FixedString)`, `Map(String)` and `Map(String)`.
+The text index also works on columns of type `Array(String)`, `Array(FixedString)`, `Map(String)` and `Map(String)`.
+
+Like for other secondary indices, each column part has its own text index.
+Furthermore, each text index is internally divided into "segments".
+The existence and size of the segments are generally transparent to users but the segment size determines the memory consumption during index construction (e.g. when two parts are merged).
+Configuration parameter `max_digestion_size_per_segment` (default: 256 MB) controls the amount of data read from the underlying column before a new segment is created.
+The default value of the parameter provides a good balance between memory usage and performance for most use cases.
+Incrementing it raises the intermediate memory consumption for index construction but also improves lookup performance since fewer segments need to be checked on average to evaluate a query.
+
+### Functions Support {#functions-support}
+
+The conditions in the `WHERE` clause contains calls of the functions that operate with columns.
+If the column is a part of an index, ClickHouse tries to use this index when performing the functions.
+ClickHouse supports different subsets of functions for the `text` index.
+
+#### equals and notEquals {#functions-example-equals-notequals}
+
+Functions `=` (equals) and `!=` (notEquals) check if the column contains rows which match the entire search term.
+Therefore, they work reasonably only with text indexes with `no_op` tokenizer.
+
+#### like, notLike and match {#functions-example-like-notlike-match}
+
+In order to use functions `like`, `notLike`, and `match` with the `text` index, the search term should be in a way that complete tokens can be extracted from it.
+
+Example:
+
+```sql
+SELECT count() FROM hackernews WHERE lower(comment) LIKE '% clickhouse support%';
+```
+
+In the example, only `clickhouse` is a complete token.
+As `support` is followed by a `%`, it could match `support`, `supports`, `supporting` etc.
+As a result, the lookup in the text index will only consider token `clickhouse`.
+
+#### startsWith and endsWith {#functions-example-startswith-endswith}
```

Comment:
Re starts/endsWith: I was wondering what happens in the index internally. It does not know the position of the token _within_ the row, hence it cannot know if the row starts or ends with the token.

Does it work in the way that the index lookup return granules for all matches (start, middle, end) and the following scan step (on the surviving granules) removes non-matching rows?

If so, we should mention that startsWith/endsWith is ineffective and currently not recommended.

Reply:
Yes, current index lookup returns all the matching granules (token at any position) and later startsWith/endsWith function itself returns the value where starts/ends with the token. I guess it's still effective to skip the non-matching granules.